lib/internal/util.js

```diff
@@ -101,13 +104,30 @@ function deprecate(fn, msg, code) {
   return deprecated;
 }
 
-function decorateErrorStack(err) {
+function decorateErrorStack(err, options) {
   if (!(isError(err) && err.stack) ||
       getHiddenValue(err, kDecoratedPrivateSymbolIndex) === true)
     return;
 
   const arrow = getHiddenValue(err, kArrowMessagePrivateSymbolIndex);
 
+  if (options?.format === 'commonjs' && arrow) {
+    const {
+      childSpecifier
+    } = options;
+    const importStatement = StringPrototypeSplit(arrow, '\n')[1];
+    const namedImports = StringPrototypeMatch(importStatement, /{.*}/)[0];
+    err.message = `The requested module '${childSpecifier}' is expected ` +
+      'to be of type CommonJS, which does not support named exports. ' +
+      'CommonJS modules can be imported by importing the default export. ' +
+      'The following code will work:\n' +
+      `import pkg from '${childSpecifier}';\n` +
+      `const ${namedImports} = pkg;`;
```

Comment:
```suggestion
      'For example:\n' +
      `import pkg from '${childSpecifier}';\n` +
      `const ${namedImports} = pkg;`;
```

I know this seems like a trivial change, but the difference is that you can't actually guarantee that 'the following code will work' because you don't know if `pkg` or the names that are being imported are possibly `const`-declared elsewhere in the file.

It would be nice to use the name of the package instead of `pkg`, if you could find an easy way to convert any string into a valid JS identifier (to handle package names like `@babel/babel-core`).

Also shouldn't `${namedImports}` be something like `{ ${namedImports.split(', ')} }`?

Reply:
namedImports is a string that is regex'd out of the failing message, not a list of imports. So it just works ™️

I don't think we should use the package name because this error can come from both a bare + non bare specifier